Styhead/rz cmn fconf

sonnguyenxg opened the pull request:
- Porting implement source code from rzv2h repo to rz-atf.
- Add fconf to support devicetree on RZ/V2H

sonnguyenxg committed:
renesas: rzv2h: Porting rzv2h implement source code.

Changes include:
	- Porting rzv2h implement source code.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
renesas: rzv2h: Add support device tree on RZ/V2H.

Changes include:
	- Add support for ATF boot using the FCONF framework
              and device tree on RZ/V2H.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

tiennguyenuh commented on plat/renesas/rz/soc/v2h/bl31_plat_setup.c:
```diff
@@ -0,0 +1,152 @@
+		return NULL;
+}
+
+void rzv2h_pass_board_id(void)
+{
+	/* Read model from QSPI for RZ/V2H*/
+	uintptr_t model_addr = (uintptr_t )(RZV2H_XSPI_MEMORY_MAP_BASE + RZV2H_BOARD_INFO_QSPI_OFFSET + OFFSET_MODEL_ID);
+	uint32_t model = mmio_read_32(model_addr);
+
+	/* Read revision id from QSPI for RZ/V2H*/
+	uintptr_t revision_addr = (uintptr_t )(RZV2H_XSPI_MEMORY_MAP_BASE + RZV2H_BOARD_INFO_QSPI_OFFSET + OFFSET_REVISION);
+	uint32_t revision = mmio_read_32(revision_addr);
+	
+	/* Get entry point info for BL33 */
+	entry_point_info_t *bl33_ep_info = bl31_plat_get_next_image_ep_info(NON_SECURE);
+
+	if (bl33_ep_info != NULL) {
+		bl33_ep_info->args.arg2 = model;
+		bl33_ep_info->args.arg3 = revision;
+	}
+}
```
We can modify `get_board_info_u32` in `board_info.c` to align with the RZ/V2H implementation, making it cleaner and easier to maintain.

sonnguyenxg replied:
Hello Tiến,
I updated.

tiennguyenuh commented on plat/renesas/rz/common/plat_storage.c:
```diff
@@ -131,37 +173,9 @@ static struct plat_io_policy policies[] = {
 				&fip_dev_handle,
 				(uintptr_t) &bl32_file_spec,
 				&open_fipdrv},
-#ifdef REMOVE_UBOOT
```
I see the remove u-boot implementation has been removed? If so, is there any reason for its removal?

sonnguyenxg replied:
Hello @Tiến,
I updated

tiennguyenuh commented on plat/renesas/rz/common/drivers/io/io_sddrv.c:
```diff
@@ -1,38 +1,31 @@
+#include <sd.h>
 
 #include "io_common.h"
-#include "io_private.h"
```
Please keep this header, as it includes the IO_SUCCESS definition. Use IO_SUCCESS instead of return 0;

sonnguyenxg replied:
fixed.

tiennguyenuh commented on plat/renesas/rz/common/bl2_plat_mem_params_desc.c:
Please keep REMOVE_UBOOT implementation

sonnguyenxg replied:
I updated.

tiennguyenuh commented on plat/renesas/rz/common/drivers/sd/esdif.h:
```diff
@@ -26,7 +26,7 @@
 /* ==== Define  ==== */
 /* ---- SD Driver work buffer ---- */
-#define SD_SIZE_OF_INIT         (640)
+// #define SD_SIZE_OF_INIT         (640)
```
If this definition is not being used, remove it entirely

sonnguyenxg replied:
fixed.

tiennguyenuh commented on plat/renesas/rz/common/drivers/sd/r_sd_cfg.h:
```diff
@@ -0,0 +1,127 @@
+	Set the method to detect card
+--------------------------------------------------------*/
+
+		/* the SD bus width */
+		#define SD_CFG_DRIVER_MODE_BUS_WIDTH    (SD_MODE_4BIT)
+
+		/* the method of check SD Status */
+		#define SD_CFG_DRIVER_MODE_METHOD       (SD_MODE_POLL)
+
+		/* the method of data transfer */
+		#define SD_CFG_DRIVER_MODE_TRANSFER     (SD_MODE_SW)
+
+		/* the card type to support */
+		#define SD_CFG_DRIVER_MODE_CARD_TYPE    (SD_MODE_MEM)
+
+		/* the speed to support */
+		#define SD_CFG_DRIVER_MODE_SPEED        (SD_MODE_DS)
+
```
Check alignment

sonnguyenxg replied:
Fixed.

tiennguyenuh commented on plat/renesas/rz/common/drivers/sd/r_sd_cfg.h:
```diff
@@ -0,0 +1,127 @@
+																						SD_CFG_DRIVER_MODE_CARD_TYPE  | \
+																						SD_CFG_DRIVER_MODE_SPEED      | \
+																						SD_CFG_DRIVER_MODE_VER)
+
```
Check alignment

sonnguyenxg replied:
fixed.

tiennguyenuh replied:
It's still not fixed. Please check it again

tiennguyenuh commented on plat/renesas/rz/common/drivers/sd/sd.h:
```diff
@@ -1,172 +1,164 @@
-                                                    /* RES_ILLEGAL_COMMAND */
-#define RES_ERASE_SEQ_ERROR         (0x10008000ul)  /* ERASE_SEQ_ERROR + WP_ERASE_SKIP */
+#define RES_SW_INTERNAL             (0xe8400000ul)              /* Driver illegal process */
+																/* OUT_OF_RANGE */
+																/* ADDRESS_ERROR */
+																/* BLOCK_LEN_ERROR */
+																/* ERASE_PARAM */
+																/* RES_ILLEGAL_COMMAND */
```
Is something missing here? What is the purpose of this comment?

sonnguyenxg replied:
Updated.

tiennguyenuh commented on plat/renesas/rz/common/plat_storage.c:
```diff
@@ -33,6 +37,22 @@ static uintptr_t sddrv_dev_handle;
 
+#if PLAT_SOC_RZV2H
+static const io_block_spec_t spirom_block_spec = {
+	.offset = RZ_SOC_SPIROM_FIP_BASE,
+	.length = RZ_SOC_SPIROM_FIP_SIZE,
+};
+
+static const io_drv_spec_t emmc_block_spec = {
+	.offset = RZ_SOC_EMMC_FIP_BASE,
+	.length = RZ_SOC_EMMC_FIP_SIZE,
+};
+
+static const io_drv_spec_t sd_block_spec = {
+	.offset = RZ_SOC_SD_FIP_BASE,
+	.length = RZ_SOC_SD_FIP_SIZE,
```
These definitions should clearly indicate they are specific to RZV2H. If they are intended solely for V2H, please replace `RZ` with `RZV2H` accordingly.

sonnguyenxg replied:
Fixed.

tiennguyenuh commented on tools/renesas/rz_boot_param/bptool.c:
Is this new boot parameter tool specifically for RZV2H? If so, we should add a description to clearly indicate that it is intended for RZV2H only

sonnguyenxg replied:
Hello Tien,
We can update it to use for all rz boards in the next version.

tiennguyenuh commented on bl2_with_dtb.hex:
I think you pushed this file by mistake

sonnguyenxg replied:
updated.

tiennguyenuh commented on tools/renesas/rz_boot_param/bptool:
I think the user can create this binary themselves; providing the .c file should be sufficient.

sonnguyenxg replied:
Hello Tien,
This tool was integrated in repo v2h.

tiennguyenuh commented on plat/renesas/rz/common/drivers/sd/r_sd_cfg.h:
```diff
@@ -0,0 +1,124 @@
+	#define SD_CFG_DRIVER_MODE_VER          (SD_MODE_VER2X)
+#else
+	/* the version to support */
+	#define SD_CFG_DRIVER_MODE_VER          (SD_MODE_VER1X)
```
I don't think these definitions need tabs.

sonnguyenxg replied:
Updated.

sonnguyenxg committed:
renesas: rzv2h: Add support board id rzv2h.

Changes include:
     - Add support board id rzv2h.
     - Update board_info to can read xSPI.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>
